drydock_provisioner/drivers/node/maasdriver/models/tag.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2017 AT&T Intellectual Property.  All other rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import drydock_provisioner.error as errors
+import drydock_provisioner.drivers.node.maasdriver.models.base as model_base
+
+import yaml
+
+class Tag(model_base.ResourceBase):
+
+    resource_url = 'tags/{resource_id}/'
+    fields = ['resource_id', 'name', 'defintion', 'kernel_opts']
+    json_fields = ['name','kernel_opts', 'comment', 'definition']
+
+    def __init__(self, api_client, **kwargs):
+        super(Tag, self).__init__(api_client, **kwargs)
+
+    def get_applied_nodes(self):
+        """
+        Query the list of nodes this tag is currently applied to
+
+        :return: List of MaaS system_ids of nodes
+        """
+
+        url = self.interpolate_url()
+
+        resp = self.api_client.get(url, op='nodes')
+
+        if resp.status_code == 200:
+            resp_json = resp.json()
+            system_id_list = []
+
+            for n in resp_json:
+                system_id_list.append(n.get('system_id'))
```

Comment:
`n.get('system_id', None)` or whatever the default value ought to be if not present